stdlib/shelve.pyi

```diff
@@ -1,35 +1,39 @@
-import collections.abc
 from _typeshed import Self
-from typing import Any, Dict, Iterator, Optional, Tuple
+from collections.abc import Iterator, MutableMapping
+from types import TracebackType
+from typing import Optional, Type, TypeVar, Union, overload
 
-class Shelf(collections.abc.MutableMapping[Any, Any]):
+_T = TypeVar("_T")
+_VT = TypeVar("_VT")
+
+class Shelf(MutableMapping[str, _VT]):
     def __init__(
-        self, dict: Dict[bytes, Any], protocol: Optional[int] = ..., writeback: bool = ..., keyencoding: str = ...
+        self, dict: MutableMapping[str, _VT], protocol: Optional[int] = ..., writeback: bool = ..., keyencoding: str = ...
     ) -> None: ...
     def __iter__(self) -> Iterator[str]: ...
     def __len__(self) -> int: ...
-    def __contains__(self, key: Any) -> bool: ...  # key should be str, but it would conflict with superclass's type signature
-    def get(self, key: str, default: Any = ...) -> Any: ...
-    def __getitem__(self, key: str) -> Any: ...
-    def __setitem__(self, key: str, value: Any) -> None: ...
+    @overload
+    def get(self, key: str) -> Optional[_VT]: ...
+    @overload
+    def get(self, key: str, default: _T = ...) -> Union[_VT, _T]: ...
```

Comment:
```suggestion
    def get(self, key: str, default: _T) -> Union[_VT, _T]: ...
```
The previous overload covers the case where default is not given.

Reply:
Done.